src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory extends StreamFactory
+{
+    /**
+     * Creates a Multipart Stream from parameters and/or files.
+     *
+     * @param array $parameters
+     * @param array $files
```

Comment:
i guess $files is a list of paths to files (or a list of streams?) that go into the multipart stream? we should explain this a bit. i guess it should be streams to be consistent - i would hope all implementations support streams.